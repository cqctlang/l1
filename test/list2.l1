x = [];
for(i = 0; i < 10000; i++)
	push(x, i);
printf("%d\n", length(x));
for(i = 10000; i > 0; i--)
	if(pop(x) != i-1)
		error("bug");
if(length(x) != 0)
	error("bug");

x = [];
for(i = 0; i < 10000; i++)
	append(x, i);
printf("%d\n", length(x));
for(i = 0; i < 10000; i++)
	if(pop(x) != i)
		error("bug");
if(length(x) != 0)
	error("bug");
