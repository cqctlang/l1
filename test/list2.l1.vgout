entry: var params, 1 locals, 1 temps
10000
10000
