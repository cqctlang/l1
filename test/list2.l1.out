entry: var params, 2 locals, 1 temps
10000
10000
